src/mixins/numeral-formatable.js

```diff
@@ -0,0 +1,278 @@
+          j > 0 && j--
+          length = 0
+        } else if (digit === '.') {
+          masked.push(this.decimal)
+          integer = true
+        }
+      }
+
+      masked.reverse()
+      // Remove leading block separator
+      this.blockSeparator.includes(masked[0]) && masked.shift()
+
+      return this.addPrefixSuffix(masked.join(''))
+    },
+    numeralZero () {
+      const zero = 0
+      this.sign = '+'
+      return this.addPrefixSuffix(zero.toFixed(this.precision).toString().replace('.', this.decimal))
+    },
+    oneChar (char, singleChar = '.') {
+      return char ? char[0] : singleChar
+    },
+    absVal (value) {
+      let parsed = parseInt(value, 10)
+
+      if (isNaN(parsed)) return 0
+      parsed = Math.abs(parsed)
+
+      // Set the general limit to that of Number.prototype.toFixed()
+      return parsed > 20 ? 20 : parsed
+    },
+    adjustNumeralCaret (selection, text) {
+      // Caret is at prefix
+      if (selection === 0) return this.numeralPrefix.length + this.sign.length
+
+      // Whether it's an integer or a fractional where abs(value) >= 10
+      if (!this.precision ||
+        (String(this.lazyValue).length > this.precision + (this.sign ? 3 : 2))) {
+        return selection
+      }
+
+      // Caret is at suffix
+      const suffixPosition = this.numeralSuffix ? text.lastIndexOf(this.numeralSuffix) : 0
+      if (suffixPosition > 0 && selection >= suffixPosition) {
+        return selection > suffixPosition ? suffixPosition
+          : this.delete || this.backspace ? suffixPosition : suffixPosition - 1
+      }
+
+      // Handle abs(value) < 10
+      // Make the caret stay where it were when the
+      // length of input is the same before and after
+      return this.backspace ? this.numeralCaretOnBackspace(selection, text)
+        : this.delete ? this.numeralCaretOnDelete(selection, text)
+          : this.numeralCaretOnInsert(selection)
+    },
+    numeralCaretOnInsert (selection) {
+      selection -= this.selection < selection ? 1 : 0
+      selection -= this.selection === selection ? 1 : 0
+
+      return selection
+    },
+    numeralCaretOnDelete (selection, text) {
+      if (text.substr(this.selection, 1) === this.decimal) {
+        selection += 1
+      } else {
+        selection += text.substr(this.numeralPrefix.length + this.sign.length, 1) === '0' ? 1 : 0
+        selection += text.substr(selection - 1, 1) === this.decimal ? 1 : 0
+      }
+
+      return selection
+    },
+    numeralCaretOnBackspace (selection, text) {
+      if (text.substr(this.numeralPrefix.length + this.sign.length, 1) === '0') {
+        // abs(value) < 1
+        text.substr(this.selection, 1) !== this.decimal && selection++
+        text.substr(this.selection - 1, 1) === this.decimal && selection++
+      }
+
+      return selection
+    },
+    // Correction on arbitrary value entered externally
+    attemptNumeralCorrection (text) {
+      const decimalPoint = text.lastIndexOf('.')
```

Comment:
Does it assume that decimal point is always a dot?

Reply:
It's always a dot in `this.lazyValue`.